### DISCUSSION TITLE
[Idea]  Add option/flag in module construct to delete module dir on upgrade

### DISCUSSION THREAD

**Discussion by madsoliver** [PrestaShop/PrestaShop#32749] [Ideas]
### The problem

Lets say i have a module that has some file `mymodule/some_file.php` 
Then I find out, I no longer need the file, so I delete the file.
Then upon uploading the new module zip file via PS Back-office, the old file is **not deleted.**

This occurs both for _any files_ but also _overrides_

The problem is greatest when it is overrides, since i would love if before needed overrides was actually deleted in `/override/` when they are deleted in `/modules/mymodule/override/`

### The solution

Adding a `$this->removeFilesOnUpgrade = true` in the module construct, the module developer could opt-in for a deletion/replacement of the current module dir when uploading a new zip-file. 

Possibly also `$this->resetOverrides = true` to check if there are overrides that came from the module that no longer are in the module dir.

Adding both these as flags make them optional.

### Alternatives

One could ask every merchant to uninstall the module with the option "delete files" and then install, but uninstalling could have side-effects not wanted. 

### Additional context

Discussed here: [](https://prestashop.slack.com/archives/C010UHP5PAR/p1685442917263009)

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by tswfi** [comment]
Somewhat related ADR draft: https://github.com/PrestaShop/ADR/pull/34 to keep module generated files out of the module folder.
